test/node-rpc-test.js

```diff
@@ -0,0 +1,230 @@
+  });
+
+  describe('getblock', function () {
+    it('should rpc getblock', async () => {
+      const {chain} = await nclient.getInfo();
+      const info = await nclient.execute('getblock', [chain.tip]);
+
+      const properties = [
+        'hash', 'confirmations', 'strippedsize',
+        'size', 'weight', 'height', 'version',
+        'versionHex', 'merkleroot', 'coinbase',
+        'tx', 'time', 'mediantime', 'nonce',
+        'bits', 'difficulty', 'chainwork',
+        'nTx', 'previousblockhash', 'nextblockhash'
+      ];
+
+      for (const property of properties)
+        assert(property in info);
+
+      assert.deepEqual(chain.height, info.height);
+      assert.deepEqual(chain.tip, info.hash);
+    });
+
+    it('should return correct height', async () => {
+      // Create an address to mine with.
+      const wallet = await node.plugins.walletdb.wdb.get(0);
+      const key = await wallet.createReceive(0);
+      const address = key.getAddress().toString(node.network.type);
+
+      // Mine two blocks.
+      await nclient.execute('generatetoaddress', [2, address]);
+
+      const {chain} = await nclient.getInfo();
+      const info = await nclient.execute('getblock', [chain.tip]);
+
+      // Assert the heights match.
+      assert.deepEqual(chain.height, info.height);
+    });
+
+    it('should return confirmations (main chain)', async () => {
+      const {chain} = await nclient.getInfo();
+
+      const {genesis} = node.network;
+      const hash = genesis.hash.toString('hex');
+
+      const info = await nclient.execute('getblock', [hash]);
+
+      assert.deepEqual(chain.height, info.confirmations - 1);
+    });
+
+    it('should return confirmations (orphan)', async () => {
+      // Get the current chain state
+      const {chain} = await nclient.getInfo();
+
+      // Get the chain entry associated with
+      // the genesis block.
+      const {genesis} = node.network;
+      let entry = await node.chain.getEntry(genesis.hash);
+
+      // Reorg from the genesis block.
+      for (let i = 0; i < chain.height + 1; i++) {
+        const block = await node.miner.mineBlock(entry);
+        await node.chain.add(block);
+        entry = await node.chain.getEntry(block.hash());
+      }
+
+      // Call getblock using the previous tip
+      const info = await nclient.execute('getblock', [chain.tip]);
+      assert.deepEqual(info.confirmations, -1);
+    });
+  });
+
+  describe('signmessagewithprivkey', function () {
+    const message = 'This is just a test message';
+    let privKeyWIF, ring;
+
+    before(async () => {
+      const addr = await wclient.execute('getnewaddress');
+      privKeyWIF = await wclient.execute('dumpprivkey', [addr]);
+      ring = KeyRing.fromSecret(privKeyWIF, 'regtest');
+    });
+
+    it('should sign message', async () => {
+      const msg = Buffer.from(MAGIC_STRING + message, 'utf8');
+      const digest = blake2b.digest(msg);
+      const expectedSignature = ring.sign(digest);
```

Comment:
I feel like hard coded values for key WIF and expected sig would be more robust tests against future regression, especially since the rpc call uses `KeyRing.fromSecret(raw, network)`, this test is comparing the same output from `KeyRing.sign()`.

Reply:
This is a good point, will push up a fix.